sonar-db/src/main/java/org/sonar/db/issue/IssueDao.java

```diff
@@ -47,27 +56,59 @@ public void selectNonClosedIssuesByModule(long componentId, ResultHandler handle
     }
   }
 
-  @CheckForNull
-  public IssueDto selectNullableByKey(DbSession session, String key) {
-    return mapper(session).selectByKey(key);
+  public Optional<IssueDto> selectByKey(DbSession session, String key) {
+    return Optional.fromNullable(mapper(session).selectByKey(key));
   }
 
-  public IssueDto selectByKey(DbSession session, String key) {
-    IssueDto issue = selectNullableByKey(session, key);
-    if (issue == null) {
-      throw new IllegalArgumentException(String.format("Issue key '%s' does not exist", key));
+  public IssueDto selectByKeyOrFail(DbSession session, String key) {
```

Comment:
In fact I think that ```selectByKeyOrFail()``` is better for code completion in IDE. It's close to ```selectByKey()``` which returns Optional.